python-frontend/src/test/java/org/sonar/python/semantic/v2/TypeInferenceV2Test.java

```diff
@@ -2329,6 +2330,20 @@ def foo():
     Assertions.assertThat(fType).isSameAs(PythonType.UNKNOWN);
   }
 
+  @Test
+  void assignmentStatementLhsTypeTest() {
+    var fileInput = inferTypes("""
+      def foo():
+          subscription[call('goal_{}'.format(1))] = 1
+      """);
+
+    var literal = TreeUtils.firstChild(fileInput, StringLiteral.class::isInstance)
+      .map(StringLiteral.class::cast)
+      .get();
+
+    var type = literal.typeV2();
+    Assertions.assertThat(type).isNotNull();
```

Comment:
I think you could simply assert that it's a string type now that the NPE is fixed.